server/sonar-web/src/main/js/components/SourceViewer/helpers/highlight.ts

```diff
@@ -95,12 +84,12 @@ function part(
  * Highlight issue locations in the list of tokens
  */
 export function highlightIssueLocations(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this function to reduce its Cognitive Complexity from 18 to the 15 allowed. ([typescript:S3776](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3776))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fpopups&issues=AWHb6qbLw6XGHKvPuQTT&open=AWHb6qbLw6XGHKvPuQTT)

[ISSUE_KEY]: AWHb6qbLw6XGHKvPuQTT